extensions/azurePublishNew/src/node/azureResources/blobStorage.ts

```diff
@@ -13,24 +28,64 @@ export const blobStorageDefinition: ResourceDefinition = {
   text: 'Azure Blob Storage',
   tier: STANDARD_LRS_TIER,
   group: AZURE_HOSTING_GROUP_NAME,
+  dependencies: [AzureResourceTypes.RESOURCE_GROUP],
+};
+
+type SkuName =
+  | 'Standard_LRS'
+  | 'Standard_GRS'
+  | 'Standard_RAGRS'
+  | 'Standard_ZRS'
+  | 'Premium_LRS'
+  | 'Premium_ZRS'
+  | 'Standard_GZRS'
+  | 'Standard_RAGZRS';
+
+export type BlobStorageConfig = ResourceConfig & {
+  key: 'blobStorage';
+  resourceGroupName: string;
+  name: string;
+  location: string;
+  skuName?: SkuName;
 };
 
-const getBlobStorageProvisionMethod = (): ProvisionMethod => {
-  return (config: BlobStorageConfigNew, workingSet: ProvisionWorkingSet): Promise<ProvisionWorkingSet> => {
-    const provisionResult = {};
+const blobStorageProvisionMethod = (provisionConfig: ProvisionServiceConfig): ProvisionMethod => {
+  const tokenCredentials = new TokenCredentials(provisionConfig.accessToken);
+  const storageManagementClient = new StorageManagementClient(tokenCredentials, provisionConfig.subscriptionId);
+
+  return async (config: BlobStorageConfig, workingSet: ProvisionWorkingSet): Promise<ProvisionWorkingSet> => {
+    try {
+      await storageManagementClient.storageAccounts.create(config.resourceGroupName, config.name, {
+        location: config.location,
+        kind: 'StorageV2',
+        sku: {
+          name: config.skuName ?? 'Standard_LRS',
+        },
+      });
+
+      const accountKeysResult = await storageManagementClient.storageAccounts.listKeys(
+        config.resourceGroupName,
+        config.name
+      );
+      const connectionString = accountKeysResult?.keys?.[0].value ?? '';
```

Comment:
This might throw, if keys.length === 0, consider changing it to 
`accountKeysResult?.keys?.[0]?.value ?? '';`

Reply:
good catch! changed.